plugins/tours/tests/phpunit/integration/renderTourComments.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Tests for render_tour_comments().
+ *
+ * @since       1.0.0
+ * @author      Robert Gadon <rgadon107>
+ * @package     spiralWebDb\CornerstoneTours\Tests\Integration
+ * @link        https://github.com/rgadon107/cornerstone
+ * @license     GNU-2.0+
+ */
+
+namespace spiralWebDb\CornerstoneTours\Tests\Integration;
+
+use spiralWebDb\Cornerstone\Tests\Integration\Test_Case;
+use function spiralWebDb\CornerstoneTours\render_tour_comments;
+
+/**
+ * @covers ::\spiralWebDb\CornerstoneTours\render_tour_comments
+ *
+ * @group   tours
+ */
+class Tests_RenderTourComments extends Test_Case {
+
+	/**
+	 * @dataProvider addTestData
+	 */
+	public function test_should_echo_meta_key_values_when_postmeta_exists( $post_data, $meta ) {
+		// Create and get the $tour_id using WordPress' factory method.
+		$tour_id = $this->factory->post->create( $post_data );
+
+		// Add post_meta to the database so we can call it.
+		add_post_meta( $tour_id, 'tour_comments', $meta['tour_comments']
+		);
+
+		$expected = (string) get_post_meta( $tour_id, 'tour_comments', true );
+
+		// Run the output buffer to fire the callback and return the output.
+		ob_start();
+		render_tour_comments( $tour_id );
+		$actual = ob_get_clean();
+
+		$this->assertSame( $expected, $actual );
+
+		// Clean up database.
+		delete_post_meta( $tour_id, 'tour_comments' );
```

Comment:
Move the `delete_post_meta()` and any clean up code to the `tearDown()` method.

Why? If the assertion fails, the test method bails out and does not run the code below the failed assertion. That means this line of code does not run if that happens. But by moving it to `tearDown()`, it will always run.

How can you do that? The `tour_id` and meta keys to delete are passed from the test data. How can the `tearDown()` know which ones to delete? Think about it. What can you do? Assign each to a property to allow methods to access this test scenarios' values. 

```php
public function test_should_echo_meta_key_values_when_postmeta_exists( $post_data, $meta ) {
	// Create and get the $tour_id using WordPress' factory method.
	$this->tour_id = $this->factory->post->create( $post_data );
	$this->meta = $meta;
```

Give it a try.